pkg/patterns/addon/pkg/cluster/cluster.go

```diff
@@ -0,0 +1,18 @@
+package cluster
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Cluster interface {
```

Comment:
controller-runtime has cluster but makes it internally. This makes me wondering whether exposing this may cause issues. Current code does not require Public interface, I can change this to private anyway.